examples/smac_smac.py

```diff
@@ -0,0 +1,194 @@
+    module = importlib.import_module(HPOLIB_SYNTH_FUNCTIONS[benchmark_name])
+    benchmark = getattr(module, benchmark_name)(rng=seed)
+    num_function_evals = benchmark.get_meta_information()['num_function_evals']
+    cs = benchmark.get_configuration_space()
+    cs.random = np.random.RandomState(seed=seed)
+    print("Run SMAC on %s with seed %d" %(benchmark.get_meta_information()["name"], seed))
+
+    print(smac_config)
+    # Build Configuration Space which defines all parameters and their ranges
+    
+    # Scenario object
+    scenario = Scenario({"run_obj": "quality",   # we optimize quality (alternatively runtime)
+                         "runcount-limit": num_function_evals,  # maximum function evaluations
+                         "cs": cs,               # configuration space
+                         "deterministic": "true",
+                         "output_dir": None
+                         })
+    
+    # string translations
+    acq = {"LCB": LCB, "EI": EI, "PI": PI}
+    r2e = {"InvScaled(y)": RunHistory2EPM4InvScaledCost, 
+           "LogScaled(y)": RunHistory2EPM4LogScaledCost,
+           "y": RunHistory2EPM4Cost}
+    init_design = {"LHD": LHDesign,
+                   "Sobol": SobolDesign, 
+                   "Default": DefaultConfiguration}
+    
+    # Optimize, using a SMAC-object
+    print("Optimizing! Depending on your machine, this might take a few minutes.")
+    smac = SMAC(scenario=scenario, 
+                rng=np.random.RandomState(seed),
+                tae_runner=wrapper,
+                initial_design=init_design[smac_config['init_design']],
+                initial_design_kwargs={'n_configs_x_params':smac_config['n_configs_x_params'],
+                                       'max_config_fracs':1.0},
+                random_configuration_chooser_kwargs={'prob': smac_config['rand_prob']},
+                runhistory2epm=r2e[smac_config['y_trans']],
+                runhistory2epm_kwargs={'scale_perc': smac_config['scale_perc']},
+                model_kwargs={'num_trees': smac_config['num_trees'],
+                              'log_y': smac_config['log_y'],
+                              'do_bootstrapping': smac_config['do_bootstrapping'],
+                              'ratio_features': smac_config['ratio_features'],
+                              'min_samples_split': smac_config['min_samples_split'],
+                              'min_samples_leaf': smac_config['min_samples_leaf']},
+                acquisition_function=acq[smac_config['acq_func']],
+                acquisition_function_kwargs={'par':smac_config['par'] if smac_config['par'] is not None else smac_config['lcb_par']}
+            )
+    
+    # hard coded for artificial functions
+    smac.solver.scenario.intensification_percentage = 1e-10
+    smac.solver.intensifier.min_chall = 1
+    smac.solver.acq_optimizer.n_sls_iterations = 10
+    smac.solver.scenario.acq_opt_challengers = 1000
+    smac.solver.predict_incumbent = True
```

Comment:
Especially this one :)

Reply:
I had the feeling from our BORF experiments that these parameters have no big impact on the performance. To make it not too hard for SMACSMAC, I would not touch it for now.